tests/test_target_database.py

```diff
@@ -1,46 +1,39 @@
-        mock_request.get(target_database._TARGET_API, status_code=404, text="Not Found")
-        target_data = target_database.get_target_data()
-        self.assertIsNone(target_data, "Target data is expected to be empty on an error")
-
-    @requests_mock.mock()
-    def test_401(self, mock_request):
-        """Given a 401 response, target data is not set."""
-        mock_request.get(target_database._TARGET_API, status_code=401)
-        target_data = target_database.get_target_data()
-        self.assertIsNone(target_data, "Target data is expected to be empty on an error")
-
-    @requests_mock.mock()
-    def test_200_invalid_json(self, mock_request):
-        """Given a valid response but invalid json, target data is not set."""
-        mock_request.get(target_database._TARGET_API, text="invalid_json")
-        target_data = target_database.get_target_data()
-        self.assertIsNone(target_data, "Target data is expected to be empty on an error")
-
-    @requests_mock.mock()
-    def test_200_no_data(self, mock_request):
-        """Given a valid response but no json data, target data is not set."""
-        mock_request.get(target_database._TARGET_API, json={})
-        target_data = target_database.get_target_data()
-        self.assertIsNone(target_data, "Target data is expected to be empty on an error")
-
-    @requests_mock.mock()
-    def test_200_value_data(self, mock_request):
+    @pytest.mark.parametrize("err_code", [404, 401])
+    def test_html_error_codes(self, requests_mock, err_code):
+        """Given an HTML error code, TargetAPIError is raised."""
+        requests_mock.get(
+            target_database._TARGET_API, status_code=err_code, text="Not Found"
+        )
+        with pytest.raises(target_database.TargetAPIError):
+            target_database.get_target_data()
+
+    def test_200_invalid_json(self, requests_mock):
+        """Given a valid response but invalid json, JSONDecodeError is raised."""
+        requests_mock.get(target_database._TARGET_API, json=None)
+        with pytest.raises(target_database.JSONDecodeError):
+            target_database.get_target_data()
+
+    def test_200_no_data_field(self, requests_mock):
+        """Given a valid response but no data field, ResponseJSONError is raised."""
+        requests_mock.get(target_database._TARGET_API, json={"notdata": []})
+        with pytest.raises(target_database.ResponseJSONError):
+            target_database.get_target_data()
+
+    def test_200_value_data(self, requests_mock):
         """Given a valid response, target data is set from the returned json."""
-        mock_request.get(target_database._TARGET_API, json={"data": 42})
+        requests_mock.get(target_database._TARGET_API, json={"data": 42})
         target_data = target_database.get_target_data()
-        self.assertEqual(42, target_data, "Target data is should match the contents of the target API data")
+        assert (
```

Comment:
Not automatically enforced, but I think we should stick to `assertion` style from unittest where possible. 😞

Reply:
Yeah I thought this may be controversial. My main reason for removing the `TestCase` inheritance was that it caused `pytest.mark.parametrize` to fail. We only parametrize one test here so I can revert to the `unittest` style if we must.

Reply:
Well, I'd be on board with not using `TestCase` as it seems unnecessary most of the time. FWIW I did raise inability to use pytest fixtures as issue with `TestCase` in the past.
https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses

@acabarbaye @madchutney could we have some input please?